torch_xla/experimental/xla_quantized_matmul.py

```diff
@@ -84,9 +106,17 @@ def quantized_matmul_xla(x: torch.Tensor,
     # Per-channel quant.
     _check_per_channel_quant_weight_dtype_shapes(x.shape[-1], scaler.shape[0],
                                                  w, scaler, zero_point)
-    out = F.linear(x, w) * scaler
+    if quantize_activation:
+      x, x_scale = _quantize_tensor(x)
+      out = torch_xla._XLAC._xla_dot_general(
+          x, w, (([-1], [-1]), ()), preferred_element_type=torch.int32)
+    else:
+      out = F.linear(x, w)
+    out = out * scaler
```

Comment:
so the output dtype will be int32?

Reply:
yes

`Matmul between x_int and w_int with int32 output dtype to avoid overflow `matmul(x_int, w_int) -> matmul_out: int32[bs, seq, out_dim]`

Reply:
The final output will be in bf16, since there will be scaler multiplying the int32 result